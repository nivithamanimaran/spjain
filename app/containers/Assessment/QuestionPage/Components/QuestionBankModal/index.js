import { makeStyles } from '@material-ui/core';
import React from 'react';
import KenDialogBox from '../../../../../components/KenDialogBox';
import QuestionBankContent from '../QuestionBankContent';

const useStyles = makeStyles(theme => ({
  mobileDialog: {
    [theme.breakpoints.down('xs')]: {
      margin: 0,
      height: '100%',
      width: '100%',
      maxHeight: '100%',
    },
    [theme.breakpoints.up('sm')]: {
      height: 'calc(100% - 64px)',
    },
  },
}));

const QuestionBankModal = props => {
  const { open, onClose } = props;
  const classes = useStyles();

  return (
    <div data-testid="questionbank-modal">
      <KenDialogBox
        open={open}
        handleClose={onClose}
        dialogActionFlag={false}
        maxWidth="987"
        styleOverrides={{ dialogPaper: classes.mobileDialog }}
      >
        <QuestionBankContent handleQuestionBankClose={onClose} />
      </KenDialogBox>
    </div>
  );
};

export default QuestionBankModal;
