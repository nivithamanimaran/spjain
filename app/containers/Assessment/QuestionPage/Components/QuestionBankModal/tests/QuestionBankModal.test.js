import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../../utils/themeProvider';
import QuestionBankModal from '../index';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});
let open = true;
const onClose = function() {
  open = false;
};
describe('Question bank modal', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QuestionBankModal open={true} onClose={onClose} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankModal open={true} onClose={onClose} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('questionbank-modal')).toMatchSnapshot();
  });

  it('Modal is closed when value of open is false', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankModal open={false} onClose={onClose} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('questionbank-modal')).toMatchSnapshot();
  });
});
