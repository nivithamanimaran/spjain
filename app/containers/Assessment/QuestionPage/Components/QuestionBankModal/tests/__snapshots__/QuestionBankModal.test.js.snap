// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question bank modal Modal is closed when value of open is false 1`] = `
<div
  data-testid="questionbank-modal"
>
  <div />
</div>
`;

exports[`Question bank modal should render and match the snapshot 1`] = `
<div
  data-testid="questionbank-modal"
>
  <div />
</div>
`;
